client/src/api/v2/camera.ts

```diff
@@ -0,0 +1,271 @@
+        value = capitalise(data.status);
+        break;
+
+      case 'diskSpaceRemaining':
+        value = formatBytes(data.diskSpaceRemaining);
+        break;
+
+      case 'recordingMinutes':
+        {
+          const totalMinutes = data.recordingMinutes as number;
+          const mins = Math.floor(totalMinutes);
+          const secs = Math.floor((totalMinutes - mins) * 60);
+          value = `${mins}m ${secs}s`;
+          name = 'Recording Time';
+        }
+        break;
+
+      default:
+        value = data[field];
+        break;
+    }
+
+    formattedData[name] = value;
+  });
+
+  return formattedData;
+}
+
+/**
+ * Initiate receiving status payloads
+ *
+ * @param subComponent The sub component to init for (e.g. recording, video feed)
+ * @param device Device
+ */
+function initStatus(subComponent: string, device?: CameraDevice) {
+  const deviceAsArr = device ? [device] : []; // only add device path if specified
+  const path = ['camera', subComponent].concat(deviceAsArr);
+  emit(`get-status-payload`, path);
+}
+
+/**
+ * Makes the device name pretty.
+ *
+ * Hardcoded for efficiency. If you don't like it, complain to Angus Trau :).
+ *
+ * @param device Device
+ * @returns  Prettied device name
+ */
+export function getPrettyDeviceName(device: CameraDevice) {
+  return device === CameraDevice.Primary ? 'Primary' : 'Secondary';
+}
+
+interface StatusPayloadOptions<T> {
+  /** Initial value for the payload */
+  initValue?: T;
+  /** Handler for updating payload */
+  payloadHandler?: (
+    setter: (p?: T) => void,
+    newPayload: T,
+    device: CameraDevice,
+  ) => T | void;
+  /** Handler for return value */
+  returnHandler?: (payload?: T, device?: CameraDevice) => T | null | undefined;
+}
+
+/**
+ * Create hooks for getting payloads
+ *
+ * @param sub Subcomponent
+ * @param opts Options
+ * @returns Hook for getting a payload
+ */
+function createStatusPayloadHook<T>(
+  sub: string,
+  {
+    initValue = undefined,
+    payloadHandler = (setter: (payload?: T) => void, newPayload: T) =>
+      setter(newPayload ?? initValue),
+    returnHandler = (payload?: T) => payload,
+  }: StatusPayloadOptions<T>,
+) {
+  return function _hook(device: CameraDevice) {
+    // Only run init once per render
+    useEffect(() => {
+      initStatus(sub, device);
+    }, [device]);
+
+    const [payload, setPayload] = useState(initValue);
+    useChannel(`status-camera-${sub}`, (newPayload: T) =>
+      payloadHandler(setPayload, newPayload, device),
+    );
+
+    return returnHandler(payload, device);
+  };
+}
+
+/**
+ * Returns the last status payload published
+ *
+ * @param device Device
+ * @returns Formatted recording status
+ */
+export const useCameraRecordingStatus = createStatusPayloadHook<
+         CameraRecordingStatusFormatted
+       >('recording', {
+         returnHandler: (payload, device) =>
+           parseRecordingPayloadData(payload?.[device as CameraDevice]),
+       });
+
+interface VideoFeedStatus {
```

Comment:
This is a nit, but I would prefer to have all interface, types, enums etc at the top of the file (even better, another file).

Reply:
@khanguslee can you recommend how you'd like the types separated to another file? (Would you like all the types to sit in one big file, or have all the `camera.ts` types in one file, split up like that?)